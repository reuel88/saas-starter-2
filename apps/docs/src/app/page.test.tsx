import { render, screen } from "@testing-library/react";
import Page from "@/app/page";

it("should render the main content with correct styles and layout", () => {
  const { baseElement } = render(<Page />);

  expect(screen.getByRole("main")).toBeInTheDocument();

  expect(baseElement).toMatchSnapshot();
});
