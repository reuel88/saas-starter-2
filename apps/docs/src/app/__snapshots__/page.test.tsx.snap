// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the main content with correct styles and layout 1`] = `
<body>
  <div>
    <main
      class="flex min-h-screen flex-col items-center justify-between p-24"
    >
      <div
        class="z-10 w-full max-w-5xl items-center justify-between font-mono text-sm lg:flex"
      >
        <p
          class="fixed left-0 top-0 flex w-full justify-center border-b border-gray-300 bg-gradient-to-b from-zinc-200 pb-6 pt-8 backdrop-blur-2xl lg:static lg:w-auto lg:rounded-xl lg:border lg:bg-gray-200  lg:p-4 dark:border-neutral-800 dark:bg-zinc-800/30 dark:from-inherit lg:dark:bg-zinc-800/30"
        >
          examples/basicÂ 
          <code
            class="font-mono font-bold"
          >
            docs
          </code>
        </p>
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 w-10 text-primary"
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          <svg
            class="lucide lucide-sun h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <path
              d="M12 2v2"
            />
            <path
              d="M12 20v2"
            />
            <path
              d="m4.93 4.93 1.41 1.41"
            />
            <path
              d="m17.66 17.66 1.41 1.41"
            />
            <path
              d="M2 12h2"
            />
            <path
              d="M20 12h2"
            />
            <path
              d="m6.34 17.66-1.41 1.41"
            />
            <path
              d="m19.07 4.93-1.41 1.41"
            />
          </svg>
          <svg
            class="lucide lucide-moon absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
            />
          </svg>
          <span
            class="sr-only"
          >
            Toggle theme
          </span>
        </button>
        <div
          class="fixed bottom-0 left-0 flex h-48 w-full items-end justify-center bg-gradient-to-t from-white via-white lg:static lg:h-auto lg:w-auto lg:bg-none dark:from-black dark:via-black"
        >
          <a
            class="pointer-events-none flex place-items-center gap-2 p-8 lg:pointer-events-auto lg:p-0"
            href="https://vercel.com?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo"
            rel="noopener noreferrer"
            target="_blank"
          >
            By
             
            <img
              alt="Vercel Logo"
              class="dark:invert"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="24"
              src="/vercel.svg"
              style="color: transparent;"
              width="100"
            />
          </a>
        </div>
      </div>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
      >
        Click me!
      </button>
      <div
        class="relative z-[-1] flex place-items-center before:absolute before:h-[300px] before:w-full before:-translate-x-1/2 before:rounded-full before:bg-gradient-radial before:from-white before:to-transparent before:blur-2xl before:content-[''] after:absolute after:-z-20 after:h-[180px] after:w-full after:translate-x-1/3 after:bg-gradient-conic after:from-sky-200 after:via-blue-200 after:blur-2xl after:content-[''] sm:before:w-[480px] sm:after:w-[240px] before:lg:h-[360px] before:dark:bg-gradient-to-br before:dark:from-transparent before:dark:to-blue-700 before:dark:opacity-10 after:dark:from-sky-900 after:dark:via-[#0141ff] after:dark:opacity-40"
      >
        <img
          alt="Next.js Logo"
          class="relative dark:drop-shadow-[0_0_0.3rem_#ffffff70] dark:invert"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="37"
          src="/next.svg"
          style="color: transparent;"
          width="180"
        />
      </div>
      <div
        class="mb-32 grid text-center lg:mb-0 lg:w-full lg:max-w-5xl lg:grid-cols-4 lg:text-left"
      >
        <a
          class="group rounded-lg border border-transparent px-5 py-4 text-primary transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
          href="https://turbo.build/repo/docs?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo""
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2
            class="mb-3 text-2xl font-semibold"
          >
            Docs
             
            <span
              class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
            >
              -&gt;
            </span>
          </h2>
          <p
            class="m-0 max-w-[30ch] text-balance text-sm opacity-50"
          >
            Find in-depth information about Turborepo features and API.
          </p>
        </a>
        <a
          class="group rounded-lg border border-transparent px-5 py-4 text-primary transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
          href="https://turbo.build/repo/docs/handbook?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo""
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2
            class="mb-3 text-2xl font-semibold"
          >
            Learn
             
            <span
              class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
            >
              -&gt;
            </span>
          </h2>
          <p
            class="m-0 max-w-[30ch] text-balance text-sm opacity-50"
          >
            Learn more about monorepos with our handbook.
          </p>
        </a>
        <a
          class="group rounded-lg border border-transparent px-5 py-4 text-primary transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
          href="https://turbo.build/repo/docs/getting-started/from-example?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo""
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2
            class="mb-3 text-2xl font-semibold"
          >
            Templates
             
            <span
              class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
            >
              -&gt;
            </span>
          </h2>
          <p
            class="m-0 max-w-[30ch] text-balance text-sm opacity-50"
          >
            Choose from over 15 examples and deploy with a single click.
          </p>
        </a>
        <a
          class="group rounded-lg border border-transparent px-5 py-4 text-primary transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
          href="https://vercel.com/new?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo""
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2
            class="mb-3 text-2xl font-semibold"
          >
            Deploy
             
            <span
              class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
            >
              -&gt;
            </span>
          </h2>
          <p
            class="m-0 max-w-[30ch] text-balance text-sm opacity-50"
          >
            Instantly deploy your Turborepo to a shareable URL with Vercel.
          </p>
        </a>
      </div>
    </main>
  </div>
</body>
`;
